pkg/libvirt/console.go

```diff
@@ -0,0 +1,216 @@
+package libvirt
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"sync"
+
+	"github.com/projecteru2/yavirt/pkg/log"
+	"github.com/projecteru2/yavirt/pkg/utils"
+	golibvirt "github.com/projecteru2/yavirt/third_party/libvirt"
+)
+
+type ConsoleFlags struct {
+	Force    bool
+	Safe     bool
+	Nonblock bool
+}
+
+func (cf *ConsoleFlags) genLibvirtFlags() (flags golibvirt.DomainConsoleFlags) {
+	if cf.Force {
+		flags |= golibvirt.DomainConsoleForce
+	}
+	if cf.Safe {
+		flags |= golibvirt.DomainConsoleSafe
+	}
+	return
+}
+
+func (cf *ConsoleFlags) genStreamFlags() (flags golibvirt.StreamFlags) {
+	if cf.Nonblock {
+		flags = golibvirt.StreamNonblock
+	}
+	return
+}
+
+type Console struct {
+	Stream *Stream
+	// pty to user
+	fromQ *utils.BytesQueue
+	// user to pty
+	toQ *utils.BytesQueue
+
+	quit chan struct{}
+	once sync.Once
```

Comment:
seems the `sync.Once` works for `quit` only, how about combines them to 
```
quit struct {
    sync.Once
    C chan struct{}
}
```
That helps to avoid abusing the `once`